From d26c5d3affe58e9b0f6b3aa0c55900efa9a15da6 Mon Sep 17 00:00:00 2001
From: Michael Biebl <biebl@debian.org>
Date: Sat, 1 Mar 2014 08:14:03 +0100
Subject: [PATCH 3/4] Enable systemd service file on installation.

This hooks up the service in bluetooth.target which is activated by systemd
when bluetooth hardware is attached. (Closes: #729808)

- Add Build-Depends on dh-systemd.
- Bump Build-Depends on cdbs to (>= 0.4.122) to get dh-systemd support.
- Workaround a bug in cdbs, which doesn't run dh_install and dh_systemd
  the correct order, by pointing dh_systemd_enable and dh_systemd_start at
  the uninstalled service file. See #715504 for details.
---
 debian/changelog | 8 ++++++++
 debian/control   | 3 ++-
 debian/rules     | 9 +++++++++
 3 files changed, 19 insertions(+), 1 deletion(-)

diff --git a/debian/changelog b/debian/changelog
index c719ec8..97bc72b 100644
--- a/debian/changelog
+++ b/debian/changelog
@@ -5,6 +5,14 @@ bluez (4.101-4.1) UNRELEASED; urgency=medium
     determined dynamically nowadays via the dependencies specified in the
     LSB header.
   * Remove obsolete and unused udev helpers for bluetoothd and hid2hci.
+  * Enable systemd service file on installation. This hooks up the service in
+    bluetooth.target which is activated by systemd when bluetooth hardware is
+    attached. (Closes: #729808)
+    - Add Build-Depends on dh-systemd.
+    - Bump Build-Depends on cdbs to (>= 0.4.122) to get dh-systemd support.
+    - Workaround a bug in cdbs, which doesn't run dh_install and dh_systemd
+      the correct order, by pointing dh_systemd_enable and dh_systemd_start at
+      the uninstalled service file. See #715504 for details.
 
  -- Michael Biebl <biebl@debian.org>  Sat, 01 Mar 2014 07:36:31 +0100
 
diff --git a/debian/control b/debian/control
index 77c1e47..c4116c3 100644
--- a/debian/control
+++ b/debian/control
@@ -3,7 +3,8 @@ Section: admin
 Priority: optional
 Maintainer: Debian Bluetooth Maintainers <pkg-bluetooth-maintainers@lists.alioth.debian.org>
 Uploaders: Nobuhiro Iwamatsu <iwamatsu@debian.org>
-Build-Depends: debhelper (>= 8.1.3), cdbs (>= 0.4.97), quilt,
+Build-Depends: debhelper (>= 8.1.3), cdbs (>= 0.4.122), quilt,
+               dh-systemd,
                autotools-dev,
                flex,
                bison,
diff --git a/debian/rules b/debian/rules
index 3ed01d5..1716eb8 100755
--- a/debian/rules
+++ b/debian/rules
@@ -8,6 +8,15 @@ include /usr/share/cdbs/1/rules/patchsys-quilt.mk
 LDFLAGS += -Wl,--as-needed
 
 DEB_DH_INSTALLINIT_ARGS = "--name=bluetooth"
+
+# Workaround a bug in cdbs, which doesn't run dh_install and dh_systemd
+# in the correct order, by pointing dh_systemd_enable and dh_systemd_start
+# at the uninstalled service file.
+# Once the bug #715504 in cdbs has been fixed, this workaround should be
+# removed again.
+DEB_DH_SYSTEMD_ENABLE_ARGS_bluez = src/bluetooth.service
+DEB_DH_SYSTEMD_START_ARGS_bluez = src/bluetooth.service
+
 DEB_CONFIGURE_EXTRA_FLAGS := \
                              --libdir=\$${prefix}/lib/$(DEB_HOST_MULTIARCH) \
                              --enable-static \
-- 
1.9.0

